tests/headless/test_endpoints.py

```diff
@@ -47,6 +49,34 @@ def test_document(base_url, is_indexed):
         assert content == 'noindex, nofollow'
 
 
+@pytest.mark.headless
+@pytest.mark.nondestructive
+def test_document_based_redirection(base_url):
+    """Ensure that content-based redirects properly redirect."""
+    url = base_url + '/en-US/docs/MDN/Promote'
```

Comment:
Here's my sequence of thoughts when reviewing this line:

"OK, this might work in production, but it may not be on staging and would need to be added to the sample database for development."

"Huh, it works in staging and development. Did I scrape that page?"

"We have a redirect link on the homepage 🤦‍♂️"

So good job, and we need to update that link on the homepage while also ensuring ``/en-US/docs/MDN/Promote`` remains in the sample DB. Future work...